APIScrape/scrape.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+require_once "../Utilities/DB.php";
+
+$db = new DB();
+$dbconnection = $db->dbConnect();
+
+/**
+ * Creates a new database into which to put the scraped data.
+ * If the database already exists, tears it down and creates a new one.
+ */
+function createDatabase(PDO $db) {
+    try {
+        $sql = "DROP DATABASE IF EXISTS `cat-test`;
+                CREATE DATABASE `cat-test`;
+                
+                use `cat-test`;
+                CREATE TABLE `breed` (
+                `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+                `breed` varchar(255) NOT NULL DEFAULT '',
+                PRIMARY KEY (`id`),
+                UNIQUE KEY `breed` (`breed`)
+                ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+               
+                CREATE TABLE `img` (
+                `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+                `img_src` varchar(255) NOT NULL DEFAULT '',
+                `breed_id` int(11) unsigned NOT NULL,
+                PRIMARY KEY (`id`),
+                UNIQUE KEY `img_src` (`img_src`)
+                ) ENGINE=InnoDB DEFAULT CHARSET=latin1;";
+        $db->exec($sql);
+        echo "Database successfully initialised.";
+    } catch (PDOException $e) {
+        echo $e->getMessage();
+    }
+}
+
+/**
+ * Make API request to get all of the breeds from theCatAPI.
+ * 
+ * @return array of breeds as 4-character breed id => breed name
+ * e.g. "ABYS" => "Abyssinian"
+ */
+function getCatBreeds():array
+{
+    $curl = curl_init();
+    curl_setopt_array($curl, array(
+        CURLOPT_URL => "https://api.thecatapi.com/v1/breeds/",
+        CURLOPT_CUSTOMREQUEST => "GET",
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_HTTPHEADER => array(
+            'x-api-key: ec254e44-3996-458b-8522-4933954d8fcd'
+        ),
+    ));
+    $response = curl_exec($curl);
+    $error = curl_error($curl);
+    curl_close($curl);
+
+    $responseArray = json_decode($response, true);
+
+    $associativeBreedArray = [];
+    for ($i = 0; $i < count($responseArray); $i++) {
+        $associativeBreedArray[$responseArray[$i]["id"]] = $responseArray[$i]["name"];
+    }
+    return $associativeBreedArray;
+}
+
+/**
+ * Puts the list of breeds into the database.
+ * 
+ * @param PDO $db - the database to insert into.
+ * @param array $catBreeds - the list of cat breeds. As breed id => breed name.
+ * @return void
+ */
+function fillCatBreedToDB(PDO $db, array $catBreeds)
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (\'' . $breed . '\');');
```

Comment:
You need to use parameter binding for safe PDO

Reply:
on it

Reply:
on it